server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/InternalComposite.java

```diff
@@ -392,16 +393,17 @@ static Object formatObject(Object obj, DocValueFormat format) {
         return obj;
     }
 
-    private static class ArrayMap extends AbstractMap<String, Object> {
+    static class ArrayMap extends AbstractMap<String, Object> implements Comparable<Map<String,Object>> {
         final List<String> keys;
-        final List<DocValueFormat> formats;
         final Object[] values;
```

Comment:
The `values` are already comparable so you could change the type to `Comparable<?>` here and then use `values[i].compareTo` ?